src/universal_transfer_operator/data_providers/dataframe/Pandasdataframe.py

```diff
@@ -49,11 +48,37 @@ def convert_dataframe_to_file(df: pd.DataFrame) -> File:
     return file
 
 
-class PandasDataframe(DataFrame):
-    """Pandas-compatible dataframe class that can be serialized and deserialized into XCom by Airflow 2.5"""
-
+class PandasdataframeDataProvider(DataframeProvider):
     version: ClassVar[int] = 1
 
+    def read(self) -> Iterator[pd.DataFrame]:
+        """Read from dataframe dataset and write to local reference locations or dataframes"""
+        yield self.dataset.dataframe
+
+    def write(self, source_ref: pd.DataFrame | DataStream) -> PandasdataframeDataProvider:
+        """Write the data to the dataframe dataset or filesystem dataset"""
+        if isinstance(source_ref, pd.DataFrame):
+            return PandasdataframeDataProvider(dataset=Dataframe(dataframe=source_ref))
+        else:
+            return PandasdataframeDataProvider(
+                dataset=Dataframe(
+                    dataframe=source_ref.actual_file.type.export_to_dataframe(
+                        stream=source_ref.remote_obj_buffer
+                    )
+                )
+            )
+
+    def equals(self, other: PandasdataframeDataProvider) -> bool:
```

Comment:
Why do we need this method? Where is this used?

Reply:
It's used in tests when comparing outputs.